lib/hawkular/inventory/entities.rb

```diff
@@ -178,39 +178,37 @@ def initialize(hash)
       @metric_id = hash[:metric_id]
     end
 
-    # rubocop:disable Metrics/CyclomaticComplexity
     def self.parse(path)
       fail 'CanonicalPath must not be nil or empty' if path.to_s.strip.length == 0
-      tmp = path.split('/')
-      hash = {}
-      tmp.each do |pair|
-        (key, val) = pair.split(';')
-        case key
-        when 't'
-          hash[:tenant_id] = val
-        when 'f'
-          hash[:feed_id] = val
-        when 'e'
-          hash[:environment_id] = val
-        when 'm'
-          hash[:metric_id] = val
-        when 'r'
-          hash[:resource_ids] = [] if hash[:resource_ids].nil?
-          hash[:resource_ids].push(val)
-        when 'mt'
-          hash[:metric_type_id] = val
-        when 'rt'
-          hash[:resource_type_id] = val
-        end
+      CanonicalPath.new(path_to_h path)
+    end
+
+    def up
```

Comment:
Sure, I'll add some doc.  I'll keep the method name the same to have parity with the java api.  I expect we can build up this ruby impl in the same way, on an as-needed basis. Commit coming...